strapi/strapi
# Add authenticationMethod and isAuthenticated properties

### What does it do?

Adds a new event called didCompleteRequest with new event properties to track: authenticationMethod and isAuthenticatedproperties

### Why is it needed?

To have insights on which method the user uses to get access to the content API. 

### How to test it?
Go to the file `packages/core/strapi/lib/services/metrics/sender.js` and change the `ANALYTICS_URI` constant to `http://localhost:4000` and run an instance of the [tracker](https://github.com/strapi/tracker) with a `console.log("data", data)` after this line ` const data = trackV2Schema.parse(ctx.request.body)` in `src/index.ts` of the tracker project. 
With all this setup, run the getStarted project and execute a query with API Tokens, you will see that the eventProperties are logged in the running tracker instance. Then do the same with a token obtained from doing a login request like: 
```
curl --location 'http://localhost:1337/api/auth/local' \
--header 'Accept: application/json' \
--header 'Content-Type: application/json' \
--data-raw '{
    "identifier": "test@test.com",
    "password": "testpass"
}'
```
You should see another log saying that the authentication method was using `users-permissions`. 
Finally, you can try to send an unauthenticated request, and you will see the `isAuthenticated` parameter set to false.
<img width="858" alt="Screenshot 2023-04-19 at 14 19 38" src="https://user-images.githubusercontent.com/16720378/233073005-1a76aa9b-e94f-442a-818b-bf5a7fff9d2d.png">

### Related issue(s)/PR(s)
